src/engine/loopingcontrol.cpp

```diff
@@ -789,20 +741,24 @@ void LoopingControl::clearActiveBeatLoop() {
     }
 }
 
-void LoopingControl::slotBeatLoop(double beats, bool keepStartPoint) {
+bool LoopingControl::currentLoopMatchesBeatloopSize() {
+    double dBeatLength;
+    LoopSamples loopSamples = m_loopSamples.getValue();
+    BpmControl::getBeatContext(m_pBeats, loopSamples.start,
+                                nullptr, nullptr, &dBeatLength, nullptr);
+    double currentLoopLength = loopSamples.end - loopSamples.start;
+    double beatLoopLength = dBeatLength * m_pCOBeatLoopSize->get();
+    return abs(currentLoopLength - beatLoopLength) < MINIMUM_AUDIBLE_LOOP_SIZE;
```

Comment:
MINIMUM_AUDIBLE_LOOP_SIZE is an arbitrary threshold, but in my testing so far it works well.